lib/openhab/core/items/item.rb

```diff
@@ -28,6 +28,31 @@ def ===(other)
         #   The item's descriptive label.
         #   @return [String, nil]
 
+        #
+        # @!attribute [rw] tags
+        #   The item's tags
+        #   @return [Array<String>]
+        #   @overload tags
+        #     Returns the item's tags.
+        #     @return [Array<String>]
+        #   @overload tags=(tags)
+        #     Sets the item's tags
+        #     @param [Array<String,Symbol,Semantics::Tag>] tags Tags to set.
+        #     @return [void]
+
+        # @!visibility private
+        def tags=(tags)
+          provider = Provider.registry.provider_for(name)
+          unless provider.is_a?(org.openhab.core.common.registry.ManagedProvider)
+            raise FrozenError, "Cannot modify item #{name} from provider #{provider.inspect}."
+          end
+
+          tags = DSL::Items::ItemBuilder.normalize_tags(*tags)
+          remove_all_tags
+          add_tags(tags)
+          provider.update(name)
```

Comment:
I'm not 100% sure if `update` here takes the item name or the object. 99% sure that the `provider_for` above takes an item name.